python/pyspark/sql/utils.py

```diff
@@ -303,3 +304,31 @@ def is_timestamp_ntz_preferred() -> bool:
     """
     jvm = SparkContext._jvm
     return jvm is not None and jvm.PythonSQLUtils.isTimestampNTZPreferred()
+
+
+@contextmanager
+def sql_conf(pairs: Dict[str, Any], *, spark: Optional["SparkSession"] = None) -> Iterator[None]:
+    """
+    A convenient context manager to set `value` to the Spark SQL configuration `key` and
+    then restores it back when it exits.
+    """
+    from pyspark.sql.session import SparkSession
+
+    assert isinstance(pairs, dict), "pairs should be a dictionary."
```

Comment:
I was thinking changing the AssertionError to ValueError. However, I decide to remain the AssertionError to avoid user-facing change for `pyspark.pandas.utils.sql_conf`.